src/core/annotation.js

```diff
@@ -227,7 +227,31 @@ function getQuadPoints(dict, rect) {
       quadPointsLists[i].push({ x, y });
     }
   }
-  return quadPointsLists;
+
+  // The PDF specification states in section 12.5.6.10 (figure 64) that the
+  // order of the quadpoints should be bottom left, bottom right, top right
+  // and top left. However, in practice PDF files use a different order,
+  // namely bottom left, bottom right, top left and top right (this is also
+  // mentioned on https://github.com/highkite/pdfAnnotate#QuadPoints), so
+  // this is the actual order we should work with. However, the situation is
+  // even worse since Adobe's own applications and other applications violate
+  // the specification and create annotations with other orders, namely top
+  // left, top right, bottom left and bottom right or even top left, top right,
+  // bottom right and bottom left. To avoid inconsistency and broken rendering,
+  // we normalize all lists to put the quadpoints in the same standard order
+  // (see https://stackoverflow.com/a/10729881).
+  return quadPointsLists.map(quadPointsList => {
+    const minX = Math.min(...quadPointsList.map(quadPoint => quadPoint.x));
```

Comment:
I think you could use something like
```js
quadPointsList.reduce(([mX, MX, mY, MY], p) => [Math.min(mX, p.x), Math.max(mY, p.y), Math.min(mY, p.y), Math.max(mY, p.y)]);
```
is slightly faster and more concise.
I discover few weeks ago that `reduce` is a way faster than `min+spread` (seef https://bugzilla.mozilla.org/show_bug.cgi?id=1674143)

Reply:
I have changed this in the new commit. Thanks!